simulation/traffic_simulator/src/utils/pose.cpp

```diff
@@ -160,5 +224,114 @@ auto boundingBoxRelativeLaneletPose(
   }
   return position;
 }
+
+/*
+  This function has been moved from pedestrian_action_node and modified,
+  in case of inconsistency please compare in original:
+  https://github.com/tier4/scenario_simulator_v2/blob/090a8d08bcb065d293a530cf641a953edf311f9f/simulation/behavior_tree_plugin/src/pedestrian/pedestrian_action_node.cpp#L67-L128
+*/
+auto estimateCanonicalizedLaneletPose(
+  const geometry_msgs::msg::Pose & map_pose,
+  const traffic_simulator_msgs::msg::BoundingBox & bounding_box,
+  const lanelet::Ids & unique_route_lanelets, const bool include_crosswalk,
+  const double matching_distance, const std::shared_ptr<hdmap_utils::HdMapUtils> & hdmap_utils_ptr)
```

Comment:
The PedestrianActionNode::estimateLaneletPose function could only be used for PedestrianEntity, but making it a free function would create the possibility of misuse.
Please emphasize that this function should only be applied to the pedestrian entity by adding the pedestrian namespace

Reply:
@hakuturu583 
I believe that introducing namespace related to entity type into `utils/pose` may result in code complexity - it will decrease its readability.

- I understand your concern, and indeed it increases the probability that `estimateCanonicalizedLaneletPose` will be used for a non-Pedestrian entity. 
- However, I think it would be a better idea to change the name to suggest that the method has an additional match rather than the classic “toCanonicalizedLaneletPose”, if “estimate” itself does not suggest it. Or just add a more informative comment.
- It is unusual here to set the position “at the end of the lanelet”. - maybe some name related to this? 

Meanwhile, I believe the method is generic enough to be worth keeping it in `utils/pose` instead of directly in `PedestrianActionNode`.

Reply:
> Meanwhile, I believe the method is generic enough to be worth keeping it in utils/pose instead of directly in PedestrianActionNode.

Yes. I agree.
I want to talk about the better name of the function `estimateCanonicalizedLaneletPose`

Reply:
How about 

- pedestrian::estimateLaneletPose
- pedestrian::computeLaneletPose
- pedestrian::transformToLaneletPose
- pedestrian::matchToLaneletPose

Reply:
I think `pedestrian::transformToLaneletPose` is best.

Reply:
@hakuturu583 
I thought of a name like `traffic_simulator::pose::toCanonicalizeLaneletPoseOrFallback`, since the `estimateCanonicalizedLaneletPose` method truly use `toCanonicalizedLaneletPose`, and if that fails, uses a fallback option - it uses hardcoded matching_distance == 2, and if that also fails, it uses `LaneletPose` at the end of the. lanelet. 

What do you think about it?

Reply:
I think `toCanonicalizeLaneletPoseOrFallback` is not a good naming because of we could not guess `Fallback to what` and `Fall back from waht` from the function's name.

Reply:
By fallback I just mean “a fallback solution when `toCanonicalizedLanePose` fails” - that is, using hardcoded `matching_distance`, and if that fails using the position at the end of the lane. “`toCanonicalizedLaneletPoseOrFallback`” tells us that `toCanonicalizedLaneletPose()` is called in the function, and if it fails fallback solutions are used. Maybe a better choice instead of “fallback” is to use “contingency” here.

I thought of two more proposals:
`pose::toCanonicalizeLaneletPoseKeepingLanelet` or `pose::toCanonicalizeLaneletPoseKeepingRoute`.

Naturally, if the names proposed by me are confusing, I understand and use the name `pedestrian::transformToLaneletPose` (more precisely `pedestrian::transformToCanonicalizedLaneletPose)`, as you proposed. :bow:

Reply:
I am a developer of this software, so I can guess `KeepingLanelet` and `KeepingRoute` means.
But if I am not a developer of this software, I think I could not guess it's meanings.
Please use `pedestrian::transformToCanonicalizedLaneletPose`

Reply:
@hakuturu583 
Of course, [done](https://github.com/tier4/scenario_simulator_v2/pull/1266/commits/6f87603e2aaaddfd8ceb08645cb3933fe3b74515#diff-1b86cbb79a60ba1de3f7e2b1918ee48d4f7b1501f7d1568b938f261ec3bae671) :bow: